src/serde_types.rs

```diff
@@ -18,6 +19,13 @@ struct DecimalVisitor;
 impl<'de> serde::de::Visitor<'de> for DecimalVisitor {
     type Value = Decimal;
 
+    fn visit_f64<E>(self, value: f64) -> Result<Decimal, E> {
+        match Decimal::from_str(&value.to_string()) {
```

Comment:
I've started promoting errors in the serde library recently. Are we able to get this changed to:

`Decimal::from_str(&value.to_string()).map_err(|_| E::invalid_value(Unexpected::Float(value), &self))`